### DISCUSSION TITLE
Offline Deal expired too fast -- not have completed sealing before deal proposal start epoch

### DISCUSSION THREAD

**Discussion by flyworker** [filecoin-project/lotus#8609] [General Q&A]
Documentation here did not say we need to specified the start epoch, however when we start sealing,it is too late.

https://docs.filecoin.io/store/lotus/very-large-files/#propose-an-offline-deal

We found the command option
--start-epoch value          specify the epoch that the deal should start at (default: -1)
but it does not tell us what is  the -1 mean

**Describe the problem**
Deal expired too fast

https://filfox.info/en/message/bafy2bzacebikhvpxget3hz55jno74llgv7nmohu4jdl2r2n6onlynqq7jh3v6

711  RecoverDealIDs  NO       NO      n/a                             1

**Sectors status**

11.	2020-12-22 06:36:40 +0000 UTC:	[event;sealing.SectorCommitFailed]	{"User":{}}
	commit check error: getting deal 1453696 for piece 0: deal 1453696 not found - deal may not have completed sealing before deal proposal start epoch, or deal may have been slashed
12.	2020-12-22 06:36:40 +0000 UTC:	[event;sealing.SectorInvalidDealIDs]	{"User":{"Return":"CommitFailed"}}
13.	2020-12-22 06:36:40 +0000 UTC:	[event;sealing.SectorUpdateDealIDs]	{"User":{"Updates":{"0":1453696}}}

**Lotus miner logs**

Please go through the logs of your miner, and include screenshots of any error-like messages you find.

Alternatively please upload full log files and share a link here

**Lotus miner diagnostic info**

Please collect the following diagnostic information, and share a link here

* lotus-miner diagnostic info `lotus-miner info all > allinfo`

** Code modifications **

If you have modified parts of lotus, please describe which areas were modified,
and the scope of those modifications

**Version**

lotus version 1.4.0+git.e9989d0e4

**Comment by dirkmc** [comment]
Thank you for the bug report.

Could you tell us the time at which
1. The deal was created by the client
2. The data was imported on the miner
3. Sealing started

It's ok if the times are approximate. It would help us to understand if there is a bug in the code or if we just need to make the documentation clearer.

**Comment by flyworker** [comment]
slack related topic https://filecoinproject.slack.com/archives/C0179RNEMU4/p1608650498126500

> Charles Cao - NBFS  13 days ago
> --start-epoch value          specify the epoch that the deal should start at (default: -1)
> the default value is -1, but what does -1 mean? (edited) 
> 
> NeonixAF f019551:blob_aww:  13 days ago
> const dealStartBufferHours uint64 = 49
> and
> 	if dealStart <= 0 { // unset, or explicitly 'epoch undefined'
> 		ts, err := a.ChainHead(ctx)
> 		if err != nil {
> 			return nil, xerrors.Errorf("failed getting chain height: %w", err)
> 		}
> 		blocksPerHour := 60 * 60 / build.BlockDelaySecs
> 		dealStart = ts.Height() + abi.ChainEpoch(dealStartBufferHours*blocksPerHour) // TODO: Get this from storage ask
> 	}
> 
> Charles Cao - NBFS  13 days ago
> so the default is dealStartBufferHours =49
> 
> NeonixAF f019551:blob_aww:  13 days ago
> I think -1 would result in the start epoch getting set to 49 hours ahead of the time the client proposes the deal

I don't think it is a bug in the code but need document to tell people who does offline transfer a longer start epoch
 is better.

This is the suggestion I gave to my client 
https://github.com/nebulaai/trusted-miner/blob/main/README.md
Thanks

**Comment by bugxzhu** [comment]
I have the same problems

**Comment by TippyFlitsUK** [comment]
Hi @flyworker

Thanks for the report. This is topic is better suited to the discussions platform.

Moving this issue to Lotus Discussions for help and troubleshooting.

Closing Lotus issue ticket.

**Comment by flyworker** [comment]
i think this issue can be closed
